### PR TITLE
feat: introduce new task redhat-dependency-analytics for RHDA v0.7.0-alpha

### PR DESCRIPTION
**Pull request by IlonaShishov**
# Changes

This pull request marks a significant milestone in the Tekton Tasks evolution. We have undergone a comprehensive refactoring process and re-branded the project from "Red Hat Codeready Dependency Analysis" to "Red Hat Dependency Analytics". 
Within this update, we've introduced a fresh Tekton Task named "redhat-dependency-analytics", aligning with the Red Hat Dependency Analytics (RHDA) platform version `0.7.0-alpha`. One of the key highlights of this update is the integration of an image utilizing the Exhort JavaScript API.
In the current version of the Red Hat Dependency Analytics Tekton Task, language support is specifically tailored to Maven and NPM. Support for GoLang and Python will be resumed in future releases.
"Red Hat Codeready Dependency Analysis" tekton task is now considered obsolete, PR to deprecation: https://github.com/tektoncd/catalog/pull/1195

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [x] Follows the [authoring recommendations][authoring]
- [x] Includes [docs][docs] (if user facing)
- [x] Includes [tests][tests] (for new tasks or changed functionality)
  - See the [end-to-end testing documentation][e2e] for guidance and CI details.
- [x] Meets the [Tekton contributor standards][contributor] (including functionality, content, code)
- [x] Commit messages follow [commit message best practices][commit]
- [x] Has a kind label. You can add one by adding a comment on this PR that
  contains `/kind <type>`. Valid types are bug, cleanup, design, documentation,
  feature, flake, misc, question, tep
- [x] Complies with [Catalog Organization TEP][TEP], see [example]. **Note** [An issue has been filed to automate this validation][validation]
  - [x] File path follows  `<kind>/<name>/<version>/name.yaml`
  - [x] Has `README.md` at `<kind>/<name>/<version>/README.md`
  - [x] Has mandatory `metadata.labels` - `app.kubernetes.io/version` the same as the `<version>` of the resource
  - [x] Has mandatory `metadata.annotations` `tekton.dev/pipelines.minVersion`
  - [x] mandatory `spec.description` follows the convention

          ```

          spec:
            description: >-
              one line summary of the resource

              Paragraph(s) to describe the resource.
          ```

_See [the contribution guide](https://github.com/tektoncd/catalog/blob/master/CONTRIBUTING.md) for more details._

[TEP]: https://github.com/tektoncd/community/blob/master/teps/0003-tekton-catalog-organization.md
[example]: https://github.com/tektoncd/catalog/tree/master/task/git-clone/0.1
[validation]: https://github.com/tektoncd/catalog/issues/413
[authoring]: https://github.com/tektoncd/catalog/blob/main/recommendations.md
[docs]: https://github.com/tektoncd/community/blob/master/standards.md#docs
[tests]: https://github.com/tektoncd/community/blob/master/standards.md#tests
[e2e]: https://github.com/tektoncd/catalog/blob/main/CONTRIBUTING.md#end-to-end-testing
[contributor]: https://github.com/tektoncd/community/blob/main/standards.md
[commit]: https://github.com/tektoncd/community/blob/master/standards.md#commit-messages

### CODE REVIEW

**Comment by piyush-garg** [comment]
We should not delete the task, it should be here and marked deprecated

we should add the new task with version 0.1

cc @vinamra28

**Comment by vinamra28** [comment]
> We should not delete the task, it should be here and marked deprecated
> 
> we should add the new task with version 0.1
> 
> cc @vinamra28

I agree to @piyush-garg, @IlonaShishov please mark the previous version as deprecated instead of deleting that

**Comment by IlonaShishov** [comment]
@vinamra28 @piyush-garg,
I made the requested changes, please review

File: task/redhat-dependency-analytics/0.1/README.md [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+==================================================
+Full report is saved into file: redhat-dependency-analytics-report.json
+Task is completed.
+```
+
+## Installation
+
+### Install task on environment using kubectl
+```
+kubectl apply -f https://raw.githubusercontent.com/tektoncd/catalog/main/task/redhat-dependency-analytics/0.7/redhat-dependency-analytics.yaml -n <NAMESPACE>
```
Comment (piyush-garg) [inline task/redhat-dependency-analytics/0.1/README.md:78]: this should be 0.1

File: task/redhat-dependency-analytics/0.1/redhat-dependency-analytics.yaml [hunk 2]
```diff
@@ -0,0 +1,54 @@
+apiVersion: tekton.dev/v1beta1
+kind: Task
+
+metadata:
+  name: redhat-dependency-analytics
+  labels:
+    app.kubernetes.io/version: "0.7"
```
Comment (piyush-garg) [inline task/redhat-dependency-analytics/0.1/redhat-dependency-analytics.yaml:7]: this should be 0.1

File: task/redhat-dependency-analytics/0.1/redhat-dependency-analytics.yaml [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+          valueFrom:
+            secretKeyRef:
+              name: exhort
+              key: EXHORT_SNYK_TOKEN
+      script: |
+        #!/bin/sh
+
+        sh /exhort.sh \
+          $(echo $(params.project-directory-path)/$(params.manifest-file-path)) \
+          $(params.output-file-path)
```
Comment (piyush-garg) [inline task/redhat-dependency-analytics/0.1/redhat-dependency-analytics.yaml:54]: please add empty line

File: task/redhat-dependency-analytics/0.1/tests/pre-apply-task-hook.sh [hunk 4]
```diff
@@ -0,0 +1,4 @@
+#!/bin/bash
+
+# Add git-clone
+add_task git-clone 0.7
```
Comment (piyush-garg) [inline task/redhat-dependency-analytics/0.1/tests/pre-apply-task-hook.sh:4]: please add empty line

File: task/redhat-dependency-analytics/0.1/README.md (hunk 1 above)
Comment (IlonaShishov) [reply task/redhat-dependency-analytics/0.1/README.md:78]: done.
Thank you

File: task/redhat-dependency-analytics/0.1/redhat-dependency-analytics.yaml (hunk 2 above)
Comment (IlonaShishov) [reply task/redhat-dependency-analytics/0.1/redhat-dependency-analytics.yaml:7]: done.
Thank you

File: task/redhat-dependency-analytics/0.1/redhat-dependency-analytics.yaml (hunk 3 above)
Comment (IlonaShishov) [reply task/redhat-dependency-analytics/0.1/redhat-dependency-analytics.yaml:54]: done.
Thank you

File: task/redhat-dependency-analytics/0.1/tests/pre-apply-task-hook.sh (hunk 4 above)
Comment (IlonaShishov) [reply task/redhat-dependency-analytics/0.1/tests/pre-apply-task-hook.sh:4]: done.
Thank you

**Review by vinamra28 — APPROVED** [review]

**Pull request merged**